src/librustc_mir/interpret/terminator.rs

```diff
@@ -264,6 +264,11 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
         match instance.def {
             ty::InstanceDef::Intrinsic(..) => {
+                if caller_abi != Abi::RustIntrinsic && caller_abi != Abi::PlatformIntrinsic {
```

Comment:
You can make this an assert. Iirc the reason we removed it was that the check is dead code when extended to cover both intrinsic ABIs. I looked a bit around the instance logic and it has such a check somewhere I think